src/global_sequences_viewer.cpp

```diff
@@ -0,0 +1,169 @@
+
+#include "global_sequences_viewer.hpp"
+
+#include <memory>
+#include <utility>
+
+#include "logging.hpp"
+#include "model.hpp"
+#include "modelversion.hpp"
+#include "statefulmodelinstance.hpp"
+#include "status.hpp"
+
+namespace ovms {
+
+static bool sequenceWatcherStarted = false;
+
+ovms::Status GlobalSequencesViewer::addVersions(std::shared_ptr<ovms::Model>& model,
+    std::shared_ptr<model_versions_t> versionsToAdd,
+    std::shared_ptr<model_versions_t> versionsFailed) {
+    for (const auto version : *versionsToAdd) {
+        if (std::count(versionsFailed->begin(), versionsFailed->end(), version))
+            continue;
+        auto modelInstance = model->getModelInstanceByVersion(version);
+        auto stetefulModelInstance = std::static_pointer_cast<StatefulModelInstance>(modelInstance);
+        std::string managerId = model->getName() + std::to_string(version);
+
+        auto status = registerManager(managerId, stetefulModelInstance->getSequenceManager().get());
+        if (status.getCode() != ovms::StatusCode::OK)
+            return status;
+    }
+
+    updateThreadInterval();
+    return ovms::StatusCode::OK;
+}
+
+ovms::Status GlobalSequencesViewer::retireVersions(std::shared_ptr<ovms::Model>& model, std::shared_ptr<model_versions_t> versionsToRetire) {
+    for (const auto version : *versionsToRetire) {
+        std::string managerId = model->getName() + std::to_string(version);
+        auto status = unregisterManager(managerId);
+        if (status.getCode() != ovms::StatusCode::OK)
+            return status;
+    }
+
+    updateThreadInterval();
+    return ovms::StatusCode::OK;
+}
+
+ovms::Status GlobalSequencesViewer::reloadVersions(std::shared_ptr<ovms::Model>& model,
+    std::shared_ptr<model_versions_t> versionsToReload,
+    std::shared_ptr<model_versions_t> versionsFailed) {
+    for (const auto version : *versionsToReload) {
+        if (std::count(versionsFailed->begin(), versionsFailed->end(), version))
+            continue;
+        std::string managerId = model->getName() + std::to_string(version);
+        auto status = unregisterManager(managerId);
+        if (status.getCode() != ovms::StatusCode::OK)
+            return status;
+
+        auto modelInstance = model->getModelInstanceByVersion(version);
+        auto stetefulModelInstance = std::static_pointer_cast<StatefulModelInstance>(modelInstance);
+
+        status = registerManager(managerId, stetefulModelInstance->getSequenceManager().get());
+        if (status.getCode() != ovms::StatusCode::OK)
+            return status;
+    }
+    updateThreadInterval();
+    return ovms::StatusCode::OK;
+}
+
+void GlobalSequencesViewer::updateThreadInterval() {
+    uint32_t lowestHalfTimeoutInterval = DEFAULT_SEQUENCE_TIMEOUT_SECONDS / 2;
+    for (auto const& [key, val] : registeredSequenceManagers) {
+        auto sequenceManager = val;
+        uint32_t newInterval = sequenceManager->getTimeout() / 2;
```

Comment:
Values of timeout of 0 and 1 will evaluate to 0 here and we won't have any interval